Dockerfile

```diff
@@ -14,30 +14,33 @@
 
 ##
 # Build stage
+# Use github action runner cached images to avoid being rate limited
+# https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images
 ## 
-FROM rust:1-buster AS build-env
+FROM buildpack-deps:buster AS build-env
 
 # Install dependencies
-RUN apt-get update && apt-get install --no-install-recommends -y build-essential clang ocl-icd-opencl-dev cmake
+RUN apt-get update && apt-get install --no-install-recommends -y build-essential clang ocl-icd-opencl-dev cmake ca-certificates curl
+RUN update-ca-certificates
+
+# Install rustup
+# https://rustup.rs/
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```

Comment:
What if we'd use ubuntu as base image for the `build-env`? It [seems](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#rust-tools) to already have `cargo` installed.

Reply:
It prompted to setup timezone when installing apt deps, not sure how to get rid of that

Reply:
Actually setting `ENV DEBIAN_FRONTEND="noninteractive"` fixes the issue, I can switch to `ubuntu:20.04` then

Reply:
So cargo is installed on gh `ubuntu` runner but not the `ubuntu` image, we still have to install `cargo` unless we don't use a multistage dockerfile

Reply:
Sounds good, thanks